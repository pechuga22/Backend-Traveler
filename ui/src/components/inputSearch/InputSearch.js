import React, { useState, useEffect } from 'react'
import {
  Header,
  Container,
  Image,
  Grid,
  Segment,
  Search,
  Form,
} from 'semantic-ui-react'
import { Link } from 'react-router-dom'
import useFetch from 'use-http'
import { useThrottle } from 'use-throttle'

import traveler from '../../static/logo-traveler.svg'
import { TourServices } from '../../services'
import Loading from '../Loading/Loading'
import Error from '../Errors/Error'
import './inputSearch.less'

export default function InputSearch({ searchInput }) {
  const [value, setValue] = useState('')

  const services = new TourServices()
  const throttledValue = useThrottle(value, 1000)
  const { url } = services.search(throttledValue)
  const { get, loading, error, data } = useFetch(url)

  useEffect(() => {
    if (!value) return
    get()
  }, [get])

  function searchChange(event) {
    let strSearchValue = event.target.value.replace(/[^a-zA-Z0-9_-]/gi, '')
    setValue(strSearchValue)
  }

  const searchTourRender = ({ title, location, image, brief, _id }) => (
    <Segment>
      <Link to={`/tours/${_id}`}>
        {loading && <Loading />}
        {error && <Error />}
        <Header as='h2' color='pink'>
          {title}
        </Header>
        <Image src={image} size='small' alt={location} floated='right' />
        <p>{brief}</p>
        <Header as='h5'>{location}</Header>
      </Link>
    </Segment>
  )

  return (
    <Container text>
      <Grid.Column align='center'>
        <Image src={traveler} centered size='small' className='logo' />
        <Form>
          <Form.Field className='search'>
            <Search
              fluid
              input={{ icon: 'search', iconPosition: 'left', ref: searchInput }}
              onKeyUp={(event) => searchChange(event)}
              placeholder='Looking for a new adventure?'
              results={data ? data.data : null}
              resultRenderer={searchTourRender}
            />
          </Form.Field>
        </Form>
      </Grid.Column>
    </Container>
  )
}
