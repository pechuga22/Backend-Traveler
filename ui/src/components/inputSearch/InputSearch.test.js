import React from 'react'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'

import InputSearch from './InputSearch'

describe('./components/Search/InputSearch.js', () => {
  describe('.render()', () => {
    it('should render InputSearch component', () => {
      const component = shallow(<InputSearch />)
      expect(toJson(component)).toMatchSnapshot()
    })

    it('render InputSearch component', () => {
      const component = mount(<InputSearch />)
      expect(component.find('.prompt').at(0).props().placeholder).toEqual(
        'Looking for a new adventure?'
      )
    })
  })
})
