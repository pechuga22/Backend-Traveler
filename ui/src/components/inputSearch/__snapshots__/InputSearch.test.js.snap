// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`./components/Search/InputSearch.js .render() should render InputSearch component 1`] = `
<Container
  text={true}
>
  <GridColumn
    align="center"
  >
    <Image
      as="img"
      centered={true}
      className="logo"
      size="small"
      src="logo-traveler.svg"
      ui={true}
    />
    <Form
      as="form"
    >
      <FormField
        className="search"
      >
        <Search
          fluid={true}
          icon="search"
          input={
            Object {
              "icon": "search",
              "iconPosition": "left",
              "ref": undefined,
            }
          }
          minCharacters={1}
          noResultsMessage="No results found."
          onKeyUp={[Function]}
          placeholder="Looking for a new adventure?"
          resultRenderer={[Function]}
          results={null}
          showNoResults={true}
        />
      </FormField>
    </Form>
  </GridColumn>
</Container>
`;
